component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java

```diff
@@ -533,14 +540,26 @@ public Response getUserAvatarById(@Context UriInfo uriInfo,
 
         if (identity.isEnable() && !identity.isDeleted()) {
           InputStream stream = identityManager.getAvatarInputStream(identity);
+          byte[] avatarContent;
           if (stream != null) {
             /*
-             * As recommended in the the RFC1341
-             * (https://www.w3.org/Protocols/rfc1341/4_Content-Type.html), we
-             * set the avatar content-type to "image/png". So, its data would be
-             * recognized as "image" by the user-agent
-             */
+            * As recommended in the the RFC1341
+            * (https://www.w3.org/Protocols/rfc1341/4_Content-Type.html), we
+            * set the avatar content-type to "image/png". So, its data would be
+            * recognized as "image" by the user-agent
+            */
             builder = Response.ok(stream, "image/png");
+            String[] dimension = size.split("x");
```

Comment:
what if size is not splittable ? 
Inspire you of function thumbnailService.parseImageSize

Reply:
It will be always splittable, the URls will specified by developers and not regular users. it should be in a specific format.

Reply:
What if a dishonest user call directly the avatar rest service with a non parsable string ?